pkg/services/ngalert/notifier/status.go

```diff
@@ -1,22 +1,23 @@
 package notifier
 
 import (
+	"context"
 	"encoding/json"
 
 	apimodels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
 )
 
 // TODO: We no longer do apimodels at this layer, move it to the API.
-func (am *alertmanager) GetStatus() apimodels.GettableStatus {
+func (am *alertmanager) GetStatus(_ context.Context) (apimodels.GettableStatus, error) {
 	config := &apimodels.PostableUserConfig{}
 	status := am.Base.GetStatus() // TODO: This should return a GettableStatus, for now it returns PostableUserConfig.
 	if status == nil {
-		return *apimodels.NewGettableStatus(&config.AlertmanagerConfig)
+		return *apimodels.NewGettableStatus(&config.AlertmanagerConfig), nil
 	}
 
 	if err := json.Unmarshal(status, config); err != nil {
-		am.logger.Error("Unable to unmarshall alertmanager config", "Err", err)
+		am.logger.Error("Unable to unmarshal alertmanager config", "err", err)
```

Comment:
Should we break out here if that fails?
```suggestion
		am.logger.Error("Unable to unmarshal alertmanager config", "err", err)
		return nil, err
```

Reply:
I didn't change the behavior of this method, it's our current GetStatus in the internal Alertmanager (introduced [here](https://github.com/grafana/grafana/pull/33227/files)). I just fixed the log lines and added a context.
I agree though, not sure why we're just ignoring the error. I'll make the method return it.